src/main/java/com/laserfiche/repository/api/RepositoryApiClientImpl.java

```diff
@@ -51,6 +54,27 @@ private <C> C createClient(Class<C> clientInterface) {
                 .create(clientInterface);
     }
 
+    @Override
+    public void setDefaultRequestHeaders(Headers defaultRequestHeaders) {
+        this.defaultHeaders = defaultRequestHeaders;
+        okBuilder.addInterceptor(chain -> {
```

Comment:
If `setDefaultRequestHeaders` is called 3 times, will it add 3 Interceptors on the okBuilder in total? And inside the interceptor, it is adding the `defaultRequestHeaders` instead of `this.defaultHeaders`, so if `setDefaultRequestHeaders` is called 3 times, will all headers from these calls be added to the request, instead of overwriting previous default request headers?

Would it make sense to add this interceptor to the constructor of `RepositoryApiClientImpl` so that it only happens once?